EventListener/UserContextSubscriber.php

```diff
@@ -87,11 +87,13 @@ public function onKernelRequest(GetResponseEvent $event)
             return;
         }
 
+        $hash = $this->hashGenerator->generateHash();
+        $event->getRequest()->attributes->set($this->hashHeader, $hash);
```

Comment:
why store this in the request and not simply as $this->hash? or could there be issues with symfony nested requests? i would expect it to instantiate a new subscriber but not sure...